src/main/java/org/spongepowered/api/service/permission/SubjectCollection.java

```diff
@@ -42,54 +58,197 @@
     String getIdentifier();
 
     /**
-     * Returns the subject specified. Will not return null.
+     * Returns a predicate which determines whether or not a given identifier is
+     * valid for a subject held by this collection.
+     *
+     * @return The predicate
+     */
+    Predicate<String> getIdentifierValidityPredicate();
+
+    /**
+     * Loads and returns the subject specified.
+     *
+     * <p>The returned future will complete exceptionally if the subject
+     * collection with the given identifier cannot be loaded.</p>
      *
-     * @param identifier The identifier to look up a subject by.
-     *                   Case-insensitive
-     * @return A stored subject if present, otherwise a subject that may be
-     *         stored if data is changed from defaults
+     * @param identifier The identifier to look up a subject by. All identifiers
+     *                   are case-insensitive
+     * @return A subject for the given identifier
+     * @throws IllegalArgumentException If the subject identifier does not pass
+     *                                  the validity predicate for this service.
      */
-    Subject get(String identifier);
+    CompletableFuture<Subject> loadSubject(String identifier);
 
     /**
-     * Returns whether there is any data stored for the given subject. The
-     * return value of this function does not influence whether or not the
-     * result from {@link #get(String)} should be queried.
+     * Returns a subject with the given identifier, if the subject is already
+     * loaded within this collection.
+     *
+     * <p>It is important to note that a subject collection with the given
```

Comment:
AFAICT this method returns subjects, why is it important to note, that a subject collection might be present or not?